[require-await] Cannot set property 'hasAwait' of null in file

**What code were you trying to parse?**
Trying to *not* parse actually. A Typescript file that is auto-generated from a swagger definition. The generated file has 
```ts
/* tslint:disable */
/* eslint-disable */
```
at the top so it shouldn't even be getting linted in the first place....

Partial code of file
```ts
getBoards(): Promise<BoardResult[]> {
    let url_ = this.baseUrl + "/boards";
    url_ = url_.replace(/[?&]$/, "");

    let options_ = <AxiosRequestConfig>{
        method: "GET",
        url: url_,
        headers: {
            "Accept": "application/json"
        }
    };

    return this.instance.request(options_).then((_response: AxiosResponse) => { 
        return this.processGetBoards(_response);
    });
}
```

The error **Cannot set property 'hasAwait' of null
Occurred while linting** references the `return this.instance.request...` line.

**What did you expect to happen?**
This file to be ignored and not throw an error while linting the project.

**What actually happened?**
Lint crashes almost immediately.

**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `2.8.0` |
| `TypeScript`                | `3.7.2` |
| `ESLint`                    | `6.6.0` |
| `node`                      | `12.4.0`  (that's what ESLint says it's using anyway) |
| `npm`                       | `6.9.1` |

If I revert back to @typescript-eslint/eslint-plugin@2.7.0 and @typescript-eslint/parser@2.7.0, everything works fine again.

Windows 10 x64 in case this could be a platform specific/path format issue.

Here is my eslint result with --debug

The file it fails on has `/* eslint-disable */` so I don't know why it's even trying to lint in the first place.

```
$ eslint --ext .js,.vue,.ts,.tsx src --debug
  eslint:cli CLI args: [ '--ext', '.js,.vue,.ts,.tsx', 'src', '--debug' ] +0ms
  eslint:cli Running on files +10ms
  eslint:ignored-paths baseDir = "D:\\MyApp\\WebClient\\ClientApp" +0ms
  eslint:ignored-paths addPatternRelativeToCwd:
  eslint:ignored-paths   original = "/node_modules/*"
  eslint:ignored-paths   cooked   = "/node_modules/*" +13ms
  eslint:ignored-paths addPatternRelativeToCwd:
  eslint:ignored-paths   original = "/bower_components/*"
  eslint:ignored-paths   cooked   = "/bower_components/*" +36ms
  eslint:ignored-paths addPatternRelativeToCwd:
  eslint:ignored-paths   original = ".*"
  eslint:ignored-paths   cooked   = ".*" +5ms
  eslint:ignored-paths addPatternRelativeToCwd:
  eslint:ignored-paths   original = "!../"
  eslint:ignored-paths   cooked   = "!../" +1ms
  eslint:ignored-paths Looking for ignore file in D:\MyApp\WebClient\ClientApp +3ms
  eslint:ignored-paths Loaded ignore file D:\MyApp\WebClient\ClientApp\.eslintignore +3ms
  eslint:ignored-paths Adding D:\MyApp\WebClient\ClientApp\.eslintignore +0ms
  eslint:ignored-paths addPatternRelativeToIgnoreFile:
  eslint:ignored-paths   original = "dist/**"
  eslint:ignored-paths   cooked   = "dist/**" +7ms
  eslint:ignored-paths addPatternRelativeToIgnoreFile:
  eslint:ignored-paths   original = ".quasar/**"
  eslint:ignored-paths   cooked   = ".quasar/**" +1ms
  eslint:ignored-paths addPatternRelativeToIgnoreFile:
  eslint:ignored-paths   original = "node_modules/**"
  eslint:ignored-paths   cooked   = "node_modules/**" +3ms
  eslint:ignored-paths addPatternRelativeToIgnoreFile:
  eslint:ignored-paths   original = "src-cordova/**"
  eslint:ignored-paths   cooked   = "src-cordova/**" +3ms
  eslint:ignored-paths addPatternRelativeToIgnoreFile:
  eslint:ignored-paths   original = "dist/**"
  eslint:ignored-paths   cooked   = "dist/**" +1ms
  eslint:ignored-paths addPatternRelativeToIgnoreFile:
  eslint:ignored-paths   original = ".quasar/**"
  eslint:ignored-paths   cooked   = ".quasar/**" +2ms
  eslint:ignored-paths addPatternRelativeToIgnoreFile:
  eslint:ignored-paths   original = "node_modules/**"
  eslint:ignored-paths   cooked   = "node_modules/**" +1ms
  eslint:ignored-paths addPatternRelativeToIgnoreFile:
  eslint:ignored-paths   original = "src-cordova/**"
  eslint:ignored-paths   cooked   = "src-cordova/**" +1ms
  eslint:ignored-paths baseDir = "D:\\MyApp\\WebClient\\ClientApp" +49ms
  eslint:ignored-paths addPatternRelativeToCwd:
  eslint:ignored-paths   original = "/node_modules/*"
  eslint:ignored-paths   cooked   = "/node_modules/*" +1ms
  eslint:ignored-paths addPatternRelativeToCwd:
  eslint:ignored-paths   original = "/bower_components/*"
  eslint:ignored-paths   cooked   = "/bower_components/*" +1ms
  eslint:ignored-paths Looking for ignore file in D:\MyApp\WebClient\ClientApp +2ms
  eslint:ignored-paths Loaded ignore file D:\MyApp\WebClient\ClientApp\.eslintignore +0ms
  eslint:ignored-paths Adding D:\MyApp\WebClient\ClientApp\.eslintignore +1ms
  eslint:ignored-paths addPatternRelativeToIgnoreFile:
  eslint:ignored-paths   original = "dist/**"
  eslint:ignored-paths   cooked   = "dist/**" +1ms
  eslint:ignored-paths addPatternRelativeToIgnoreFile:
  eslint:ignored-paths   original = ".quasar/**"
  eslint:ignored-paths   cooked   = ".quasar/**" +1ms
  eslint:ignored-paths addPatternRelativeToIgnoreFile:
  eslint:ignored-paths   original = "node_modules/**"
  eslint:ignored-paths   cooked   = "node_modules/**" +2ms
  eslint:ignored-paths addPatternRelativeToIgnoreFile:
  eslint:ignored-paths   original = "src-cordova/**"
  eslint:ignored-paths   cooked   = "src-cordova/**" +1ms
  eslint:ignored-paths addPatternRelativeToIgnoreFile:
  eslint:ignored-paths   original = "dist/**"
  eslint:ignored-paths   cooked   = "dist/**" +1ms
  eslint:ignored-paths addPatternRelativeToIgnoreFile:
  eslint:ignored-paths   original = ".quasar/**"
  eslint:ignored-paths   cooked   = ".quasar/**" +2ms
  eslint:ignored-paths addPatternRelativeToIgnoreFile:
  eslint:ignored-paths   original = "node_modules/**"
  eslint:ignored-paths   cooked   = "node_modules/**" +1ms
  eslint:ignored-paths addPatternRelativeToIgnoreFile:
  eslint:ignored-paths   original = "src-cordova/**"
  eslint:ignored-paths   cooked   = "src-cordova/**" +1ms
  eslint:file-enumerator Start to iterate files: [ 'src' ] +0ms
  eslint:file-enumerator Directory: D:\MyApp\WebClient\ClientApp\src +1ms
  eslint:ignored-paths contains: +8ms
  eslint:ignored-paths   target   = "D:\\MyApp\\WebClient\\ClientApp\\src\\" +0ms
  eslint:ignored-paths   base     = "D:\\MyApp\\WebClient\\ClientApp" +1ms
  eslint:ignored-paths   relative = "src\\" +0ms
  eslint:ignored-paths   result   = false +1ms
  eslint:file-enumerator Enter the directory: D:\MyApp\WebClient\ClientApp\src +5ms
  eslint:ignored-paths contains: +1ms
  eslint:ignored-paths   target   = "D:\\MyApp\\WebClient\\ClientApp\\src\\api-client\\" +1ms
  eslint:ignored-paths   base     = "D:\\MyApp\\WebClient\\ClientApp" +1ms
  eslint:ignored-paths   relative = "src\\api-client\\" +0ms
  eslint:ignored-paths   result   = false +1ms
  eslint:file-enumerator Enter the directory: D:\MyApp\WebClient\ClientApp\src\api-client +4ms
  eslint:cascading-config-array-factory Load config files for D:\MyApp\WebClient\ClientApp\src\api-client. +0ms
  eslint:cascading-config-array-factory No cache found: D:\MyApp\WebClient\ClientApp\src\api-client. +1ms
  eslint:config-array-factory Config file not found on D:\MyApp\WebClient\ClientApp\src\api-client +0ms
  eslint:cascading-config-array-factory No cache found: D:\MyApp\WebClient\ClientApp\src. +55ms
  eslint:config-array-factory Config file not found on D:\MyApp\WebClient\ClientApp\src +54ms
  eslint:cascading-config-array-factory No cache found: D:\MyApp\WebClient\ClientApp. +1ms
  eslint:config-array-factory Loading JS config file: D:\MyApp\WebClient\ClientApp\.eslintrc.js +2ms
  eslint:config-array-factory Config file found: D:\MyApp\WebClient\ClientApp\.eslintrc.js +4ms
  eslint:config-array-factory Loading {extends:"eslint:recommended"} relative to D:\MyApp\WebClient\ClientApp\.eslintrc.js +25ms
  eslint:config-array-factory Loading JS config file: D:\MyApp\WebClient\ClientApp\node_modules\eslint\conf\eslint-recommended.js +1ms
  eslint:config-array-factory Loading {extends:"plugin:import/errors"} relative to D:\MyApp\WebClient\ClientApp\.eslintrc.js +67ms
  eslint:config-array-factory Loading plugin "import" from D:\MyApp\WebClient\ClientApp\.eslintrc.js +1ms
  eslint:config-array-factory Loaded: eslint-plugin-import@2.18.2 (D:\MyApp\WebClient\ClientApp\node_modules\eslint-plugin-import\lib\index.js) +4ms
  eslint:config-array-factory Plugin D:\MyApp\WebClient\ClientApp\node_modules\eslint-plugin-import\lib\index.js loaded in: 229ms +229ms
  eslint:config-array-factory Loading plugin "import" from D:\MyApp\WebClient\ClientApp\node_modules\eslint-plugin-import\lib\index.js +2ms
  eslint:config-array-factory Loaded: eslint-plugin-import@2.18.2 (D:\MyApp\WebClient\ClientApp\node_modules\eslint-plugin-import\lib\index.js) +1ms
  eslint:config-array-factory Plugin D:\MyApp\WebClient\ClientApp\node_modules\eslint-plugin-import\lib\index.js loaded in: 0ms +57ms
  eslint:config-array-factory Loading {extends:"plugin:import/warnings"} relative to D:\MyApp\WebClient\ClientApp\.eslintrc.js +2ms
  eslint:config-array-factory Loading plugin "import" from D:\MyApp\WebClient\ClientApp\.eslintrc.js +0ms
  eslint:config-array-factory Loaded: eslint-plugin-import@2.18.2 (D:\MyApp\WebClient\ClientApp\node_modules\eslint-plugin-import\lib\index.js) +1ms
  eslint:config-array-factory Plugin D:\MyApp\WebClient\ClientApp\node_modules\eslint-plugin-import\lib\index.js loaded in: 0ms +0ms
  eslint:config-array-factory Loading plugin "import" from D:\MyApp\WebClient\ClientApp\node_modules\eslint-plugin-import\lib\index.js +1ms
  eslint:config-array-factory Loaded: eslint-plugin-import@2.18.2 (D:\MyApp\WebClient\ClientApp\node_modules\eslint-plugin-import\lib\index.js) +1ms
  eslint:config-array-factory Plugin D:\MyApp\WebClient\ClientApp\node_modules\eslint-plugin-import\lib\index.js loaded in: 0ms +1ms
  eslint:config-array-factory Loading {extends:"plugin:import/typescript"} relative to D:\MyApp\WebClient\ClientApp\.eslintrc.js +0ms
  eslint:config-array-factory Loading plugin "import" from D:\MyApp\WebClient\ClientApp\.eslintrc.js +1ms
  eslint:config-array-factory Loaded: eslint-plugin-import@2.18.2 (D:\MyApp\WebClient\ClientApp\node_modules\eslint-plugin-import\lib\index.js) +1ms
  eslint:config-array-factory Plugin D:\MyApp\WebClient\ClientApp\node_modules\eslint-plugin-import\lib\index.js loaded in: 0ms +0ms
  eslint:config-array-factory Loading {extends:"plugin:@typescript-eslint/eslint-recommended"} relative to D:\MyApp\WebClient\ClientApp\.eslintrc.js +1ms
  eslint:config-array-factory Loading plugin "@typescript-eslint" from D:\MyApp\WebClient\ClientApp\.eslintrc.js +1ms
  eslint:config-array-factory Loaded: @typescript-eslint/eslint-plugin@2.8.0 (D:\MyApp\WebClient\ClientApp\node_modules\@typescript-eslint\eslint-plugin\dist\index.js) +3ms
  eslint:config-array-factory Plugin D:\MyApp\WebClient\ClientApp\node_modules\@typescript-eslint\eslint-plugin\dist\index.js loaded in: 399ms +401ms
  eslint:config-array-factory Loading {extends:"plugin:@typescript-eslint/recommended"} relative to D:\MyApp\WebClient\ClientApp\.eslintrc.js +3ms
  eslint:config-array-factory Loading plugin "@typescript-eslint" from D:\MyApp\WebClient\ClientApp\.eslintrc.js +63ms
  eslint:config-array-factory Loaded: @typescript-eslint/eslint-plugin@2.8.0 (D:\MyApp\WebClient\ClientApp\node_modules\@typescript-eslint\eslint-plugin\dist\index.js) +2ms
  eslint:config-array-factory Plugin D:\MyApp\WebClient\ClientApp\node_modules\@typescript-eslint\eslint-plugin\dist\index.js loaded in: 0ms +0ms
  eslint:config-array-factory Loading {extends:"./configs/base.json"} relative to D:\MyApp\WebClient\ClientApp\node_modules\@typescript-eslint\eslint-plugin\dist\index.js +1ms
  eslint:config-array-factory package.json was not found: Cannot find module './configs/base.json/package.json'
Require stack:
- D:\MyApp\WebClient\ClientApp\node_modules\@typescript-eslint\eslint-plugin\dist\index.js +1ms
  eslint:config-array-factory Loaded: ./configs/base.json (D:\MyApp\WebClient\ClientApp\node_modules\@typescript-eslint\eslint-plugin\dist\configs\base.json) +2ms
  eslint:config-array-factory Loading JSON config file: D:\MyApp\WebClient\ClientApp\node_modules\@typescript-eslint\eslint-plugin\dist\configs\base.json +0ms
  eslint:config-array-factory Loading parser "@typescript-eslint/parser" from D:\MyApp\WebClient\ClientApp\node_modules\@typescript-eslint\eslint-plugin\dist\configs\base.json +3ms
  eslint:config-array-factory Loaded: @typescript-eslint/parser@2.8.0 (D:\MyApp\WebClient\ClientApp\node_modules\@typescript-eslint\parser\dist\parser.js) +5ms
  eslint:config-array-factory Loading plugin "@typescript-eslint" from D:\MyApp\WebClient\ClientApp\node_modules\@typescript-eslint\eslint-plugin\dist\configs\base.json +88ms
  eslint:config-array-factory Loaded: @typescript-eslint/eslint-plugin@2.8.0 (D:\MyApp\WebClient\ClientApp\node_modules\@typescript-eslint\eslint-plugin\dist\index.js) +1ms
  eslint:config-array-factory Plugin D:\MyApp\WebClient\ClientApp\node_modules\@typescript-eslint\eslint-plugin\dist\index.js loaded in: 0ms +1ms
  eslint:config-array-factory Loading {extends:"plugin:@typescript-eslint/recommended-requiring-type-checking"} relative to D:\MyApp\WebClient\ClientApp\.eslintrc.js +1ms
  eslint:config-array-factory Loading plugin "@typescript-eslint" from D:\MyApp\WebClient\ClientApp\.eslintrc.js +0ms
  eslint:config-array-factory Loaded: @typescript-eslint/eslint-plugin@2.8.0 (D:\MyApp\WebClient\ClientApp\node_modules\@typescript-eslint\eslint-plugin\dist\index.js) +1ms
  eslint:config-array-factory Plugin D:\MyApp\WebClient\ClientApp\node_modules\@typescript-eslint\eslint-plugin\dist\index.js loaded in: 0ms +1ms
  eslint:config-array-factory Loading {extends:"./configs/base.json"} relative to D:\MyApp\WebClient\ClientApp\node_modules\@typescript-eslint\eslint-plugin\dist\index.js +0ms
  eslint:config-array-factory package.json was not found: Cannot find module './configs/base.json/package.json'
Require stack:
- D:\MyApp\WebClient\ClientApp\node_modules\@typescript-eslint\eslint-plugin\dist\index.js +1ms
  eslint:config-array-factory Loaded: ./configs/base.json (D:\MyApp\WebClient\ClientApp\node_modules\@typescript-eslint\eslint-plugin\dist\configs\base.json) +1ms
  eslint:config-array-factory Loading JSON config file: D:\MyApp\WebClient\ClientApp\node_modules\@typescript-eslint\eslint-plugin\dist\configs\base.json +1ms
  eslint:config-array-factory Loading parser "@typescript-eslint/parser" from D:\MyApp\WebClient\ClientApp\node_modules\@typescript-eslint\eslint-plugin\dist\configs\base.json +2ms
  eslint:config-array-factory Loaded: @typescript-eslint/parser@2.8.0 (D:\MyApp\WebClient\ClientApp\node_modules\@typescript-eslint\parser\dist\parser.js) +1ms
  eslint:config-array-factory Loading plugin "@typescript-eslint" from D:\MyApp\WebClient\ClientApp\node_modules\@typescript-eslint\eslint-plugin\dist\configs\base.json +1ms
  eslint:config-array-factory Loaded: @typescript-eslint/eslint-plugin@2.8.0 (D:\MyApp\WebClient\ClientApp\node_modules\@typescript-eslint\eslint-plugin\dist\index.js) +0ms
  eslint:config-array-factory Plugin D:\MyApp\WebClient\ClientApp\node_modules\@typescript-eslint\eslint-plugin\dist\index.js loaded in: 0ms +1ms
  eslint:config-array-factory Loading {extends:"plugin:vue/essential"} relative to D:\MyApp\WebClient\ClientApp\.eslintrc.js +0ms
  eslint:config-array-factory Loading plugin "vue" from D:\MyApp\WebClient\ClientApp\.eslintrc.js +1ms
  eslint:config-array-factory Loaded: eslint-plugin-vue@6.0.1 (D:\MyApp\WebClient\ClientApp\node_modules\eslint-plugin-vue\lib\index.js) +3ms
  eslint:config-array-factory Plugin D:\MyApp\WebClient\ClientApp\node_modules\eslint-plugin-vue\lib\index.js loaded in: 324ms +324ms
  eslint:config-array-factory Loading {extends:"D:\\MyApp\\WebClient\\ClientApp\\node_modules\\eslint-plugin-vue\\lib\\configs\\base.js"} relative to D:\MyApp\WebClient\ClientApp\node_modules\eslint-plugin-vue\lib\index.js +1ms
  eslint:config-array-factory package.json was not found: Cannot find module 'D:\MyApp\WebClient\ClientApp\node_modules\eslint-plugin-vue\lib\configs\base.js/package.json'
Require stack:
- D:\MyApp\WebClient\ClientApp\node_modules\eslint-plugin-vue\lib\index.js +1ms
  eslint:config-array-factory Loaded: D:\MyApp\WebClient\ClientApp\node_modules\eslint-plugin-vue\lib\configs\base.js (D:\MyApp\WebClient\ClientApp\node_modules\eslint-plugin-vue\lib\configs\base.js) +145ms
  eslint:config-array-factory Loading JS config file: D:\MyApp\WebClient\ClientApp\node_modules\eslint-plugin-vue\lib\configs\base.js +2ms
  eslint:config-array-factory Loading parser "D:\\MyApp\\WebClient\\ClientApp\\node_modules\\vue-eslint-parser\\index.js" from D:\MyApp\WebClient\ClientApp\node_modules\eslint-plugin-vue\lib\configs\base.js +3ms
  eslint:config-array-factory package.json was not found: Cannot find module 'D:\MyApp\WebClient\ClientApp\node_modules\vue-eslint-parser\index.js/package.json'
Require stack:
- D:\MyApp\WebClient\ClientApp\node_modules\eslint-plugin-vue\lib\configs\base.js +1ms
  eslint:config-array-factory Loaded: D:\MyApp\WebClient\ClientApp\node_modules\vue-eslint-parser\index.js (D:\MyApp\WebClient\ClientApp\node_modules\vue-eslint-parser\index.js) +1ms   
  eslint:config-array-factory Loading plugin "vue" from D:\MyApp\WebClient\ClientApp\node_modules\eslint-plugin-vue\lib\configs\base.js +1ms
  eslint:config-array-factory Loaded: eslint-plugin-vue@6.0.1 (D:\MyApp\WebClient\ClientApp\node_modules\eslint-plugin-vue\lib\index.js) +0ms
  eslint:config-array-factory Plugin D:\MyApp\WebClient\ClientApp\node_modules\eslint-plugin-vue\lib\index.js loaded in: 0ms +1ms
  eslint:config-array-factory Loading plugin "import" from D:\MyApp\WebClient\ClientApp\.eslintrc.js +1ms
  eslint:config-array-factory Loaded: eslint-plugin-import@2.18.2 (D:\MyApp\WebClient\ClientApp\node_modules\eslint-plugin-import\lib\index.js) +1ms
  eslint:config-array-factory Plugin D:\MyApp\WebClient\ClientApp\node_modules\eslint-plugin-import\lib\index.js loaded in: 0ms +0ms
  eslint:config-array-factory Loading plugin "@typescript-eslint" from D:\MyApp\WebClient\ClientApp\.eslintrc.js +1ms
  eslint:config-array-factory Loaded: @typescript-eslint/eslint-plugin@2.8.0 (D:\MyApp\WebClient\ClientApp\node_modules\@typescript-eslint\eslint-plugin\dist\index.js) +1ms
  eslint:config-array-factory Plugin D:\MyApp\WebClient\ClientApp\node_modules\@typescript-eslint\eslint-plugin\dist\index.js loaded in: 0ms +0ms
  eslint:config-array-factory Loading plugin "vue" from D:\MyApp\WebClient\ClientApp\.eslintrc.js +0ms
  eslint:config-array-factory Loaded: eslint-plugin-vue@6.0.1 (D:\MyApp\WebClient\ClientApp\node_modules\eslint-plugin-vue\lib\index.js) +1ms
  eslint:config-array-factory Plugin D:\MyApp\WebClient\ClientApp\node_modules\eslint-plugin-vue\lib\index.js loaded in: 0ms +1ms
  eslint:cascading-config-array-factory Stop traversing because of 'root:true'. +1s
  eslint:rules Loading rule 'constructor-super' (remaining=269) +0ms
  eslint:rules Loading rule 'for-direction' (remaining=268) +4ms
  eslint:rules Loading rule 'getter-return' (remaining=267) +1ms
  eslint:rules Loading rule 'no-async-promise-executor' (remaining=266) +10ms
  eslint:rules Loading rule 'no-case-declarations' (remaining=265) +91ms
  eslint:rules Loading rule 'no-class-assign' (remaining=264) +3ms
  eslint:rules Loading rule 'no-compare-neg-zero' (remaining=263) +2ms
  eslint:rules Loading rule 'no-cond-assign' (remaining=262) +2ms
  eslint:rules Loading rule 'no-const-assign' (remaining=261) +3ms
  eslint:rules Loading rule 'no-constant-condition' (remaining=260) +2ms
  eslint:rules Loading rule 'no-control-regex' (remaining=259) +3ms
  eslint:rules Loading rule 'no-debugger' (remaining=258) +73ms
  eslint:rules Loading rule 'no-delete-var' (remaining=257) +1ms
  eslint:rules Loading rule 'no-dupe-args' (remaining=256) +2ms
  eslint:rules Loading rule 'no-dupe-class-members' (remaining=255) +1ms
  eslint:rules Loading rule 'no-dupe-keys' (remaining=254) +2ms
  eslint:rules Loading rule 'no-duplicate-case' (remaining=253) +2ms
  eslint:rules Loading rule 'no-empty' (remaining=252) +1ms
  eslint:rules Loading rule 'no-empty-character-class' (remaining=251) +4ms
  eslint:rules Loading rule 'no-empty-pattern' (remaining=250) +2ms
  eslint:rules Loading rule 'no-ex-assign' (remaining=249) +1ms
  eslint:rules Loading rule 'no-extra-boolean-cast' (remaining=248) +1ms
  eslint:rules Loading rule 'no-extra-semi' (remaining=247) +2ms
  eslint:rules Loading rule 'no-fallthrough' (remaining=246) +2ms
  eslint:rules Loading rule 'no-func-assign' (remaining=245) +3ms
  eslint:rules Loading rule 'no-global-assign' (remaining=244) +1ms
  eslint:rules Loading rule 'no-inner-declarations' (remaining=243) +4ms
  eslint:rules Loading rule 'no-invalid-regexp' (remaining=242) +2ms
  eslint:rules Loading rule 'no-irregular-whitespace' (remaining=241) +3ms
  eslint:rules Loading rule 'no-misleading-character-class' (remaining=240) +4ms
  eslint:rules Loading rule 'no-mixed-spaces-and-tabs' (remaining=239) +10ms
  eslint:rules Loading rule 'no-new-symbol' (remaining=238) +75ms
  eslint:rules Loading rule 'no-obj-calls' (remaining=237) +2ms
  eslint:rules Loading rule 'no-octal' (remaining=236) +1ms
  eslint:rules Loading rule 'no-prototype-builtins' (remaining=235) +2ms
  eslint:rules Loading rule 'no-redeclare' (remaining=234) +2ms
  eslint:rules Loading rule 'no-regex-spaces' (remaining=233) +3ms
  eslint:rules Loading rule 'no-self-assign' (remaining=232) +2ms
  eslint:rules Loading rule 'no-shadow-restricted-names' (remaining=231) +3ms
  eslint:rules Loading rule 'no-sparse-arrays' (remaining=230) +1ms
  eslint:rules Loading rule 'no-this-before-super' (remaining=229) +2ms
  eslint:rules Loading rule 'no-undef' (remaining=228) +2ms
  eslint:rules Loading rule 'no-unexpected-multiline' (remaining=227) +2ms
  eslint:rules Loading rule 'no-unreachable' (remaining=226) +2ms
  eslint:rules Loading rule 'no-unsafe-finally' (remaining=225) +2ms
  eslint:rules Loading rule 'no-unsafe-negation' (remaining=224) +1ms
  eslint:rules Loading rule 'no-unused-labels' (remaining=223) +3ms
  eslint:rules Loading rule 'no-unused-vars' (remaining=222) +65ms
  eslint:rules Loading rule 'no-useless-catch' (remaining=221) +69ms
  eslint:rules Loading rule 'no-useless-escape' (remaining=220) +2ms
  eslint:rules Loading rule 'no-with' (remaining=219) +2ms
  eslint:rules Loading rule 'require-atomic-updates' (remaining=218) +4ms
  eslint:rules Loading rule 'require-yield' (remaining=217) +2ms
  eslint:rules Loading rule 'use-isnan' (remaining=216) +3ms
  eslint:rules Loading rule 'valid-typeof' (remaining=215) +4ms
  eslint:rules Loading rule 'camelcase' (remaining=214) +10ms
  eslint:rules Loading rule 'no-array-constructor' (remaining=213) +21ms
  eslint:rules Loading rule 'no-empty-function' (remaining=212) +52ms
  eslint:rules Loading rule 'no-use-before-define' (remaining=211) +9ms
  eslint:rules Loading rule 'no-var' (remaining=210) +14ms
  eslint:rules Loading rule 'prefer-const' (remaining=209) +5ms
  eslint:rules Loading rule 'prefer-rest-params' (remaining=208) +3ms
  eslint:rules Loading rule 'prefer-spread' (remaining=207) +64ms
  eslint:rules Loading rule 'require-await' (remaining=206) +4ms
  eslint:rules Loading rule 'generator-star-spacing' (remaining=205) +17ms
  eslint:rules Loading rule 'arrow-parens' (remaining=204) +3ms
  eslint:rules Loading rule 'comma-dangle' (remaining=203) +2ms
  eslint:rules Loading rule 'one-var' (remaining=202) +4ms
  eslint:rules Loading rule 'max-len' (remaining=201) +3ms
  eslint:rules Loading rule 'no-trailing-spaces' (remaining=200) +73ms
  eslint:rules Loading rule 'indent' (remaining=199) +3ms
  eslint:rules Loading rule 'no-console' (remaining=198) +1ms
  eslint:rules Loading rule 'prefer-promise-reject-errors' (remaining=197) +2ms
  eslint:rules Loading rule 'quotes' (remaining=196) +1ms
  eslint:rules Loading rule 'semi' (remaining=195) +3ms
  eslint:cascading-config-array-factory Configuration was determined: ConfigArray [ { name: '.eslintrc.js » eslint:recommended', filePath: 'D:\\MyApp\\WebClient\\ClientApp\\node_modules\\eslint\\conf\\eslint-recommended.js', criteria: null, env: undefined, globals: undefined, noInlineConfig: undefined, parser: undefined, parserOptions: undefined, plugins: undefined, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: { 'constructor-super': 'error', 'for-direction': 'error', 'getter-return': 'error', 'no-async-promise-executor': 'error', 'no-case-declarations': 'error', 'no-class-assign': 'error', 'no-compare-neg-zero': 'error', 'no-cond-assign': 'error', 'no-const-assign': 'error', 'no-constant-condition': 'error', 'no-control-regex': 'error', 'no-debugger': 'error', 'no-delete-var': 'error', 'no-dupe-args': 'error', 'no-dupe-class-members': 'error', 'no-dupe-keys': 'error', 'no-duplicate-case': 'error', 'no-empty': 'error', 'no-empty-character-class': 'error', 'no-empty-pattern': 'error', 'no-ex-assign': 'error', 'no-extra-boolean-cast': 'error', 'no-extra-semi': 'error', 'no-fallthrough': 'error', 'no-func-assign': 'error', 'no-global-assign': 'error', 
'no-inner-declarations': 'error', 'no-invalid-regexp': 'error', 'no-irregular-whitespace': 'error', 'no-misleading-character-class': 'error', 'no-mixed-spaces-and-tabs': 'error', 'no-new-symbol': 'error', 'no-obj-calls': 'error', 'no-octal': 'error', 'no-prototype-builtins': 'error', 'no-redeclare': 'error', 'no-regex-spaces': 'error', 'no-self-assign': 'error', 'no-shadow-restricted-names': 'error', 'no-sparse-arrays': 'error', 'no-this-before-super': 'error', 'no-undef': 'error', 'no-unexpected-multiline': 'error', 'no-unreachable': 'error', 'no-unsafe-finally': 'error', 'no-unsafe-negation': 'error', 'no-unused-labels': 'error', 'no-unused-vars': 'error', 'no-useless-catch': 'error', 'no-useless-escape': 'error', 'no-with': 'error', 'require-atomic-updates': 'error', 'require-yield': 'error', 'use-isnan': 'error', 'valid-typeof': 'error' }, settings: undefined }, { name: '.eslintrc.js » plugin:import/errors', filePath: 'D:\\MyApp\\WebClient\\ClientApp\\node_modules\\eslint-plugin-import\\lib\\index.js', criteria: null, env: undefined, globals: undefined, noInlineConfig: undefined, parser: undefined, parserOptions: undefined, plugins: { import: [Object] }, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: { 'import/no-unresolved': 2, 'import/named': 2, 'import/namespace': 2, 'import/default': 2, 'import/export': 2 }, settings: undefined }, { name: '.eslintrc.js » plugin:import/warnings', filePath: 'D:\\MyApp\\WebClient\\ClientApp\\node_modules\\eslint-plugin-import\\lib\\index.js', criteria: null, env: undefined, globals: undefined, noInlineConfig: undefined, parser: undefined, parserOptions: undefined, plugins: { import: [Object] }, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: { 'import/no-named-as-default': 1, 'import/no-named-as-default-member': 1, 'import/no-duplicates': 1 }, settings: undefined }, { name: '.eslintrc.js » plugin:import/typescript', filePath: 'D:\\MyApp\\WebClient\\ClientApp\\node_modules\\eslint-plugin-import\\lib\\index.js', criteria: null, env: undefined, globals: undefined, noInlineConfig: undefined, parser: undefined, parserOptions: undefined, plugins: undefined, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: undefined, settings: { 'import/extensions': [Array], 'import/parsers': [Object], 'import/resolver': [Object] } }, { name: '.eslintrc.js » plugin:@typescript-eslint/eslint-recommended', filePath: 'D:\\MyApp\\WebClient\\ClientApp\\node_modules\\@typescript-eslint\\eslint-plugin\\dist\\index.js', criteria: null, env: undefined, globals: undefined, noInlineConfig: undefined, parser: undefined, parserOptions: undefined, plugins: undefined, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: undefined, settings: undefined }, { name: '.eslintrc.js » plugin:@typescript-eslint/eslint-recommended#overrides[0]', filePath: 'D:\\MyApp\\WebClient\\ClientApp\\node_modules\\@typescript-eslint\\eslint-plugin\\dist\\index.js', criteria: { includes: [Array], excludes: null, basePath: 'D:\\MyApp\\WebClient\\ClientApp' }, env: undefined, globals: undefined, noInlineConfig: undefined, parser: undefined, parserOptions: undefined, plugins: undefined, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: { 'getter-return': 'off', 'no-dupe-args': 'off', 'no-dupe-keys': 'off', 'no-unreachable': 'off', 'valid-typeof': 'off', 'no-const-assign': 'off', 'no-new-symbol': 'off', 'no-this-before-super': 'off', 'no-undef': 'off', 'no-dupe-class-members': 'off', 'no-redeclare': 'off' }, settings: undefined }, { name: '.eslintrc.js » plugin:@typescript-eslint/recommended » ./configs/base.json', filePath: 'D:\\MyApp\\WebClient\\ClientApp\\node_modules\\@typescript-eslint\\eslint-plugin\\dist\\configs\\base.json', criteria: null, env: undefined, globals: undefined, noInlineConfig: undefined, parser: { error: null, filePath: 'D:\\MyApp\\WebClient\\ClientApp\\node_modules\\@typescript-eslint\\parser\\dist\\parser.js', id: '@typescript-eslint/parser', importerName: '.eslintrc.js » plugin:@typescript-eslint/recommended » ./configs/base.json', importerPath: 'D:\\MyApp\\WebClient\\ClientApp\\node_modules\\@typescript-eslint\\eslint-plugin\\dist\\configs\\base.json' }, parserOptions: { sourceType: 'module' }, plugins: { '@typescript-eslint': [Object] }, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: undefined, settings: undefined }, { name: '.eslintrc.js » plugin:@typescript-eslint/recommended', filePath: 'D:\\MyApp\\WebClient\\ClientApp\\node_modules\\@typescript-eslint\\eslint-plugin\\dist\\index.js', criteria: null, env: undefined, globals: undefined, noInlineConfig: undefined, parser: undefined, parserOptions: undefined, plugins: undefined, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: { '@typescript-eslint/adjacent-overload-signatures': 'error', '@typescript-eslint/ban-ts-ignore': 'error', '@typescript-eslint/ban-types': 'error', camelcase: 'off', '@typescript-eslint/camelcase': 'error', '@typescript-eslint/class-name-casing': 'error', '@typescript-eslint/consistent-type-assertions': 'error', '@typescript-eslint/explicit-function-return-type': 'warn', '@typescript-eslint/interface-name-prefix': 'error', '@typescript-eslint/member-delimiter-style': 'error', 'no-array-constructor': 'off', '@typescript-eslint/no-array-constructor': 'error', 'no-empty-function': 'off', '@typescript-eslint/no-empty-function': 'error', '@typescript-eslint/no-empty-interface': 'error', '@typescript-eslint/no-explicit-any': 'warn', '@typescript-eslint/no-inferrable-types': 'error', '@typescript-eslint/no-misused-new': 'error', '@typescript-eslint/no-namespace': 'error', '@typescript-eslint/no-non-null-assertion': 'warn', '@typescript-eslint/no-this-alias': 'error', 'no-unused-vars': 'off', '@typescript-eslint/no-unused-vars': 'warn', 'no-use-before-define': 'off', '@typescript-eslint/no-use-before-define': 'error', '@typescript-eslint/no-var-requires': 'error', '@typescript-eslint/prefer-namespace-keyword': 'error', '@typescript-eslint/triple-slash-reference': 'error', '@typescript-eslint/type-annotation-spacing': 'error', 'no-var': 'error', 'prefer-const': 'error', 'prefer-rest-params': 'error', 'prefer-spread': 'error' }, settings: undefined }, { 
name: '.eslintrc.js » plugin:@typescript-eslint/recommended-requiring-type-checking » ./configs/base.json', filePath: 'D:\\MyApp\\WebClient\\ClientApp\\node_modules\\@typescript-eslint\\eslint-plugin\\dist\\configs\\base.json', criteria: null, env: undefined, globals: undefined, noInlineConfig: undefined, parser: { error: null, filePath: 'D:\\MyApp\\WebClient\\ClientApp\\node_modules\\@typescript-eslint\\parser\\dist\\parser.js', id: '@typescript-eslint/parser', importerName: '.eslintrc.js » plugin:@typescript-eslint/recommended-requiring-type-checking » ./configs/base.json', importerPath: 'D:\\MyApp\\WebClient\\ClientApp\\node_modules\\@typescript-eslint\\eslint-plugin\\dist\\configs\\base.json' }, parserOptions: { sourceType: 'module' }, plugins: { '@typescript-eslint': [Object] }, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: undefined, settings: undefined }, { name: '.eslintrc.js » plugin:@typescript-eslint/recommended-requiring-type-checking', filePath: 'D:\\MyApp\\WebClient\\ClientApp\\node_modules\\@typescript-eslint\\eslint-plugin\\dist\\index.js', criteria: null, env: undefined, globals: undefined, noInlineConfig: undefined, parser: undefined, parserOptions: undefined, plugins: undefined, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: { '@typescript-eslint/await-thenable': 'error', '@typescript-eslint/no-for-in-array': 'error', '@typescript-eslint/no-misused-promises': 'error', '@typescript-eslint/no-unnecessary-type-assertion': 'error', '@typescript-eslint/prefer-includes': 'error', '@typescript-eslint/prefer-regexp-exec': 'error', '@typescript-eslint/prefer-string-starts-ends-with': 'error', 'require-await': 'off', '@typescript-eslint/require-await': 'error', '@typescript-eslint/unbound-method': 'error', 'no-var': 'error', 'prefer-const': 'error', 'prefer-rest-params': 'error', 'prefer-spread': 'error' }, settings: undefined }, { name: '.eslintrc.js » plugin:vue/essential » D:\\MyApp\\WebClient\\ClientApp\\node_modules\\eslint-plugin-vue\\lib\\configs\\base.js#processors["vue/.vue"]', filePath: 'D:\\MyApp\\WebClient\\ClientApp\\node_modules\\eslint-plugin-vue\\lib\\configs\\base.js', criteria: { includes: [Array], excludes: null, basePath: 'D:\\MyApp\\WebClient\\ClientApp' }, env: undefined, globals: undefined, noInlineConfig: undefined, parser: undefined, parserOptions: undefined, plugins: undefined, processor: 'vue/.vue', reportUnusedDisableDirectives: undefined, root: undefined, rules: undefined, settings: undefined }, { name: '.eslintrc.js » plugin:vue/essential » D:\\MyApp\\WebClient\\ClientApp\\node_modules\\eslint-plugin-vue\\lib\\configs\\base.js', filePath: 'D:\\MyApp\\WebClient\\ClientApp\\node_modules\\eslint-plugin-vue\\lib\\configs\\base.js', criteria: null, env: { browser: true, es6: true }, globals: undefined, noInlineConfig: undefined, parser: { error: null, filePath: 'D:\\MyApp\\WebClient\\ClientApp\\node_modules\\vue-eslint-parser\\index.js', id: 'D:\\MyApp\\WebClient\\ClientApp\\node_modules\\vue-eslint-parser\\index.js', importerName: '.eslintrc.js » plugin:vue/essential » D:\\MyApp\\WebClient\\ClientApp\\node_modules\\eslint-plugin-vue\\lib\\configs\\base.js', importerPath: 'D:\\MyApp\\WebClient\\ClientApp\\node_modules\\eslint-plugin-vue\\lib\\configs\\base.js' }, parserOptions: { ecmaVersion: 2018, sourceType: 'module', ecmaFeatures: [Object] }, plugins: { vue: [Object] }, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: { 'vue/comment-directive': 'error', 'vue/jsx-uses-vars': 'error' }, settings: undefined }, { name: '.eslintrc.js » plugin:vue/essential', filePath: 'D:\\MyApp\\WebClient\\ClientApp\\node_modules\\eslint-plugin-vue\\lib\\index.js', criteria: null, env: undefined, globals: undefined, noInlineConfig: undefined, parser: undefined, parserOptions: undefined, plugins: undefined, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: { 'vue/no-async-in-computed-properties': 'error', 'vue/no-dupe-keys': 'error', 'vue/no-duplicate-attributes': 'error', 'vue/no-parsing-error': 'error', 'vue/no-reserved-keys': 'error', 'vue/no-shared-component-data': 'error', 'vue/no-side-effects-in-computed-properties': 'error', 'vue/no-template-key': 'error', 'vue/no-textarea-mustache': 'error', 'vue/no-unused-components': 'error', 'vue/no-unused-vars': 'error', 'vue/no-use-v-if-with-v-for': 'error', 'vue/require-component-is': 'error', 'vue/require-prop-type-constructor': 'error', 'vue/require-render-return': 'error', 'vue/require-v-for-key': 'error', 'vue/require-valid-default-prop': 'error', 'vue/return-in-computed-property': 'error', 'vue/use-v-on-exact': 'error', 'vue/valid-template-root': 'error', 'vue/valid-v-bind': 'error', 'vue/valid-v-cloak': 'error', 'vue/valid-v-else-if': 'error', 'vue/valid-v-else': 'error', 'vue/valid-v-for': 'error', 'vue/valid-v-html': 'error', 'vue/valid-v-if': 'error', 'vue/valid-v-model': 'error', 'vue/valid-v-on': 'error', 'vue/valid-v-once': 'error', 'vue/valid-v-pre': 'error', 'vue/valid-v-show': 'error', 'vue/valid-v-text': 'error' }, settings: undefined }, { name: '.eslintrc.js#processors["vue/.vue"]', filePath: 'D:\\MyApp\\WebClient\\ClientApp\\.eslintrc.js', criteria: { includes: [Array], excludes: null, basePath: 'D:\\MyApp\\WebClient\\ClientApp' }, env: undefined, globals: undefined, noInlineConfig: undefined, parser: undefined, parserOptions: undefined, plugins: undefined, processor: 'vue/.vue', reportUnusedDisableDirectives: undefined, root: undefined, rules: undefined, settings: undefined }, { name: '.eslintrc.js', filePath: 'D:\\MyApp\\WebClient\\ClientApp\\.eslintrc.js', criteria: null, env: { node: true }, globals: { ga: true, cordova: true, __statics: true, process: true }, noInlineConfig: undefined, parser: undefined, parserOptions: { extraFileExtensions: [Array], parser: '@typescript-eslint/parser', project: './tsconfig.json', sourceType: 'module' }, plugins: { import: [Object], '@typescript-eslint': [Object], vue: [Object] }, processor: undefined, reportUnusedDisableDirectives: undefined, root: 
true, rules: { 'generator-star-spacing': 'off', 'arrow-parens': 'off', 'comma-dangle': [Array], 'one-var': 'off', 'import/first': 'off', 'import/named': 'error', 'import/namespace': 'error', 'import/default': 'error', 'import/export': 'error', 'import/extensions': 'off', 'import/no-unresolved': 'off', 'import/no-extraneous-dependencies': 'off', 'max-len': [Array], 'no-trailing-spaces': 'error', indent: 
'off', 'no-console': 'off', 'no-debugger': 'off', 'no-extra-semi': 'off', 'no-unused-vars': 'off', 'prefer-promise-reject-errors': 'off', quotes: 'error', semi: 'off', '@typescript-eslint/explicit-function-return-type': 'off', '@typescript-eslint/explicit-member-accessibility': [Array], '@typescript-eslint/indent': [Array], '@typescript-eslint/interface-name-prefix': 'off', '@typescript-eslint/member-delimiter-style': [Array], '@typescript-eslint/no-explicit-any': 'off', '@typescript-eslint/no-floating-promises': [Array], '@typescript-eslint/no-unused-vars': [Array], '@typescript-eslint/semi': [Array] 
}, settings: undefined }, { name: '.eslintrc.js#overrides[0]', filePath: 'D:\\MyApp\\WebClient\\ClientApp\\.eslintrc.js', criteria: { includes: [Array], excludes: null, basePath: 'D:\\MyApp\\WebClient\\ClientApp' }, env: undefined, globals: undefined, noInlineConfig: undefined, parser: undefined, parserOptions: undefined, plugins: undefined, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: { '@typescript-eslint/explicit-member-accessibility': 'off' }, settings: undefined } ] on D:\MyApp\WebClient\ClientApp\src\api-client +820ms
  eslint:ignored-paths contains: +3s
  eslint:ignored-paths   target   = "D:\\MyApp\\WebClient\\ClientApp\\src\\api-client\\index.ts" +68ms
  eslint:ignored-paths   base     = "D:\\MyApp\\WebClient\\ClientApp" +3ms
  eslint:ignored-paths   relative = "src\\api-client\\index.ts" +1ms
  eslint:ignored-paths   result   = false +0ms
  eslint:file-enumerator Yield: index.ts +3s
  eslint:cli-engine Lint D:\MyApp\WebClient\ClientApp\src\api-client\index.ts +0ms
  eslint:linter Linting code for D:\MyApp\WebClient\ClientApp\src\api-client\index.ts (pass 1) +0ms
  eslint:linter Verify +0ms
  eslint:linter With ConfigArray: D:\MyApp\WebClient\ClientApp\src\api-client\index.ts +1ms
  eslint:config-comment-parser Parsing list config +0ms
  eslint:linter An error occurred while traversing +8s
  eslint:linter Filename: D:\MyApp\WebClient\ClientApp\src\api-client\index.ts +4ms
  eslint:linter Line: 308 +0ms
  eslint:linter Parser Options: { ecmaFeatures: { globalReturn: false, jsx: true },
  ecmaVersion: 9,
  extraFileExtensions: [ '.vue' ],
  parser: '@typescript-eslint/parser',
  project: './tsconfig.json',
  sourceType: 'module' } +1ms
  eslint:linter Parser Path: D:\MyApp\WebClient\ClientApp\node_modules\vue-eslint-parser\index.js +2ms
  eslint:linter Settings: { 'import/extensions': [ '.ts', '.tsx', '.d.ts', '.js', '.jsx' ],
  'import/parsers': { '@typescript-eslint/parser': [ '.ts', '.tsx', '.d.ts' ] },
  'import/resolver': { node: { extensions: [Array] } } } +1ms
TypeError: Cannot set property 'hasAwait' of null
Occurred while linting D:\MyApp\WebClient\ClientApp\src\api-client\index.ts:308
    at Object.AwaitExpression (D:\MyApp\WebClient\ClientApp\node_modules\eslint\lib\rules\require-await.js:92:36)
    at ReturnStatement (D:\MyApp\WebClient\ClientApp\node_modules\@typescript-eslint\eslint-plugin\dist\rules\require-await.js:68:27)
    at listeners.(anonymous function).forEach.listener (D:\MyApp\WebClient\ClientApp\node_modules\eslint\lib\linter\safe-emitter.js:45:58)
    at Array.forEach (<anonymous>)
    at Object.emit (D:\MyApp\WebClient\ClientApp\node_modules\eslint\lib\linter\safe-emitter.js:45:38)
    at NodeEventGenerator.applySelector (D:\MyApp\WebClient\ClientApp\node_modules\eslint\lib\linter\node-event-generator.js:254:26)
    at NodeEventGenerator.applySelectors (D:\MyApp\WebClient\ClientApp\node_modules\eslint\lib\linter\node-event-generator.js:283:22)
    at NodeEventGenerator.enterNode (D:\MyApp\WebClient\ClientApp\node_modules\eslint\lib\linter\node-event-generator.js:297:14)
    at CodePathAnalyzer.enterNode (D:\MyApp\WebClient\ClientApp\node_modules\eslint\lib\linter\code-path-analysis\code-path-analyzer.js:634:23)
    at nodeQueue.forEach.traversalInfo (D:\MyApp\WebClient\ClientApp\node_modules\eslint\lib\linter\linter.js:936:32)
```

--- a/packages/eslint-plugin/src/rules/require-await.ts
+++ b/packages/eslint-plugin/src/rules/require-await.ts
@@ -42,13 +42,12 @@ export default util.createRule<Options, MessageIds>({
     }
 
     return {
-      'FunctionDeclaration[async = true]': rules.FunctionDeclaration,
-      'FunctionExpression[async = true]': rules.FunctionExpression,
+      FunctionDeclaration: rules.FunctionDeclaration,
+      FunctionExpression: rules.FunctionExpression,
+      ArrowFunctionExpression: rules.ArrowFunctionExpression,
       'ArrowFunctionExpression[async = true]'(
         node: TSESTree.ArrowFunctionExpression,
       ): void {
-        rules.ArrowFunctionExpression(node);
-
         // If body type is not BlockStatment, we need to check the return type here
         if (node.body.type !== AST_NODE_TYPES.BlockStatement) {
           const expression = parserServices.esTreeNodeToTSNodeMap.get(
@@ -56,15 +55,13 @@ export default util.createRule<Options, MessageIds>({
           );
           if (expression && isThenableType(expression)) {
             // tell the base rule to mark the scope as having an await so it ignores it
-            rules.AwaitExpression(node as never);
+            rules.AwaitExpression();
           }
         }
       },
-      'FunctionDeclaration[async = true]:exit':
-        rules['FunctionDeclaration:exit'],
-      'FunctionExpression[async = true]:exit': rules['FunctionExpression:exit'],
-      'ArrowFunctionExpression[async = true]:exit':
-        rules['ArrowFunctionExpression:exit'],
+      'FunctionDeclaration:exit': rules['FunctionDeclaration:exit'],
+      'FunctionExpression:exit': rules['FunctionExpression:exit'],
+      'ArrowFunctionExpression:exit': rules['ArrowFunctionExpression:exit'],
       AwaitExpression: rules.AwaitExpression,
       ForOfStatement: rules.ForOfStatement,
 
@@ -74,7 +71,7 @@ export default util.createRule<Options, MessageIds>({
         >(node);
         if (expression && isThenableType(expression)) {
           // tell the base rule to mark the scope as having an await so it ignores it
-          rules.AwaitExpression(node as never);
+          rules.AwaitExpression();
         }
       },
     };

--- a/packages/eslint-plugin/tests/rules/require-await.test.ts
+++ b/packages/eslint-plugin/tests/rules/require-await.test.ts
@@ -28,106 +28,87 @@ const noAwaitAsyncFunctionExpression: any = {
 
 ruleTester.run('require-await', rule, {
   valid: [
-    {
-      // Non-async function declaration
-      code: `function numberOne(): number {
-        return 1;
-      }`,
-    },
-    {
-      // Non-async function expression
-      code: `const numberOne = function(): number {
-        return 1;
-      }`,
-    },
-    {
-      // Non-async arrow function expression (concise-body)
-      code: `const numberOne = (): number => 1;`,
-    },
-    {
-      // Non-async arrow function expression (block-body)
-      code: `const numberOne = (): number => {
-        return 1;
-      };`,
-    },
-    {
-      // Async function declaration with await
-      code: `async function numberOne(): Promise<number> {
-        return await 1;
-      }`,
-    },
-    {
-      // Async function expression with await
-      code: `const numberOne = async function(): Promise<number> {
-        return await 1;
-      }`,
-    },
-    {
-      // Async arrow function expression with await (concise-body)
-      code: `const numberOne = async (): Promise<number> => await 1;`,
-    },
-    {
-      // Async arrow function expression with await (block-body)
-      code: `const numberOne = async (): Promise<number> => {
-        return await 1;
-      };`,
-    },
-    {
-      // Async function declaration with promise return
-      code: `async function numberOne(): Promise<number> {
-        return Promise.resolve(1);
-      }`,
-    },
-    {
-      // Async function expression with promise return
-      code: `const numberOne = async function(): Promise<number> {
-        return Promise.resolve(1);
-      }`,
-    },
-    {
-      // Async arrow function with promise return (concise-body)
-      code: `const numberOne = async (): Promise<number> => Promise.resolve(1);`,
-    },
-    {
-      // Async arrow function with promise return (block-body)
-      code: `const numberOne = async (): Promise<number> => {
-        return Promise.resolve(1);
-      };`,
-    },
-    {
-      // Async function declaration with async function return
-      code: `async function numberOne(): Promise<number> {
-        return getAsyncNumber(1);
-      }
-      async function getAsyncNumber(x: number): Promise<number> {
-        return Promise.resolve(x);
-      }`,
-    },
-    {
-      // Async function expression with async function return
-      code: `const numberOne = async function(): Promise<number> {
-        return getAsyncNumber(1);
-      }
-      const getAsyncNumber = async function(x: number): Promise<number> {
-        return Promise.resolve(x);
-      }`,
-    },
-    {
-      // Async arrow function with async function return (concise-body)
-      code: `const numberOne = async (): Promise<number> => getAsyncNumber(1);
-      const getAsyncNumber = async function(x: number): Promise<number> {
-        return Promise.resolve(x);
-      }`,
-    },
-    {
-      // Async arrow function with async function return (block-body)
-      code: `const numberOne = async (): Promise<number> => {
-        return getAsyncNumber(1);
-      };
-      const getAsyncNumber = async function(x: number): Promise<number> {
-        return Promise.resolve(x);
-      }`,
-    },
+    // Non-async function declaration
+    `function numberOne(): number {
+      return 1;
+    }`,
+    // Non-async function expression
+    `const numberOne = function(): number {
+      return 1;
+    }`,
+    // Non-async arrow function expression (concise-body)
+    `const numberOne = (): number => 1;`,
+    // Non-async arrow function expression (block-body)
+    `const numberOne = (): number => {
+      return 1;
+    };`,
+    // Non-async function that returns a promise
+    // https://github.com/typescript-eslint/typescript-eslint/issues/1226
+    `
+function delay() {
+  return Promise.resolve();
+}
+    `,
+    `
+const delay = () => {
+  return Promise.resolve();
+}
+    `,
+    `const delay = () => Promise.resolve();`,
+    // Async function declaration with await
+    `async function numberOne(): Promise<number> {
+      return await 1;
+    }`,
+    // Async function expression with await
+    `const numberOne = async function(): Promise<number> {
+      return await 1;
+    }`,
+    // Async arrow function expression with await (concise-body)
+    `const numberOne = async (): Promise<number> => await 1;`,
+    // Async arrow function expression with await (block-body)
+    `const numberOne = async (): Promise<number> => {
+      return await 1;
+    };`,
+    // Async function declaration with promise return
+    `async function numberOne(): Promise<number> {
+      return Promise.resolve(1);
+    }`,
+    // Async function expression with promise return
+    `const numberOne = async function(): Promise<number> {
+      return Promise.resolve(1);
+    }`,
+    // Async arrow function with promise return (concise-body)
+    `const numberOne = async (): Promise<number> => Promise.resolve(1);`,
+    // Async arrow function with promise return (block-body)
+    `const numberOne = async (): Promise<number> => {
+      return Promise.resolve(1);
+    };`,
+    // Async function declaration with async function return
+    `async function numberOne(): Promise<number> {
+      return getAsyncNumber(1);
+    }
+    async function getAsyncNumber(x: number): Promise<number> {
+      return Promise.resolve(x);
+    }`,
+    // Async function expression with async function return
+    `const numberOne = async function(): Promise<number> {
+      return getAsyncNumber(1);
+    }
+    const getAsyncNumber = async function(x: number): Promise<number> {
+      return Promise.resolve(x);
+    }`,
+    // Async arrow function with async function return (concise-body)
+    `const numberOne = async (): Promise<number> => getAsyncNumber(1);
+    const getAsyncNumber = async function(x: number): Promise<number> {
+      return Promise.resolve(x);
+    }`,
+    // Async arrow function with async function return (block-body)
+    `const numberOne = async (): Promise<number> => {
+      return getAsyncNumber(1);
+    };
+    const getAsyncNumber = async function(x: number): Promise<number> {
+      return Promise.resolve(x);
+    }`,
     // https://github.com/typescript-eslint/typescript-eslint/issues/1188
     `
 async function testFunction(): Promise<void> {

--- a/packages/eslint-plugin/typings/eslint-rules.d.ts
+++ b/packages/eslint-plugin/typings/eslint-rules.d.ts
@@ -451,7 +451,7 @@ declare module 'eslint/lib/rules/require-await' {
         node: TSESTree.ArrowFunctionExpression,
       ): void;
       ReturnStatement(node: TSESTree.ReturnStatement): void;
-      AwaitExpression(node: TSESTree.AwaitExpression): void;
+      AwaitExpression(): void;
       ForOfStatement(node: TSESTree.ForOfStatement): void;
     }
   >;

